lib/trino-parquet/src/main/java/io/trino/parquet/ParquetTypeUtils.java

```diff
@@ -354,14 +355,37 @@ public static Optional<Field> constructField(Type type, ColumnIO columnIO)
             return Optional.of(new GroupField(type, repetitionLevel, definitionLevel, required, ImmutableList.of(keyField, valueField)));
         }
         if (type instanceof ArrayType arrayType) {
-            GroupColumnIO groupColumnIO = (GroupColumnIO) columnIO;
-            if (groupColumnIO.getChildrenCount() != 1) {
-                return Optional.empty();
+            // Per the parquet spec (https://github.com/apache/parquet-format/blob/master/LogicalTypes.md):
+            // `A repeated field that is neither contained by a LIST- or MAP-annotated group nor annotated by LIST or MAP should be interpreted as a required list of required elements
+            // where the element type is the type of the field.`
+            //
+            // A parquet encoding for a required list of strings can be expressed in two ways, however for backwards compatibility they should be handled the same, so here we need
+            // to adjust repetition and definition levels when converting ColumnIOs to Fields.
+            //      1. required group colors (LIST) {
+            //              repeated group list {
+            //                  required string element;
+            //              }
+            //         }
+            //      2. repeated binary colors (STRING);
+            if (required && columnIO instanceof PrimitiveColumnIO && repetitionLevel > 0 && definitionLevel > 0) {
```

Comment:
Also requires `columnIO.getType().getRepetition() == REPEATED`

Reply:
Yeah, mistakenly replaced the repeated check you described with the `required` check, but should be the opposite.